local_gemma/modeling_local_gemma_2.py

```diff
@@ -169,4 +186,23 @@ def from_pretrained(
             # for consistent behaviour with bitsandbytes, we move the model to the device always
             model.to(device, dtype=preset_kwargs["torch_dtype"])
 
+        if torch_compile and device != "cuda":
+            raise ValueError(
+                "Torch compile is only compatible with cuda devices. Set `torch_compile=False` in `.from_pretrained`"
+                f"for device {device}."
+            )
+        elif torch_compile:
+            model = fuse_attention_weights(model, device, torch_dtype)
+            model.forward = torch.compile(model.forward, mode="reduce-overhead", fullgraph=True)
```

Comment:
Would be curious to hear whether you agree with this design @gante. I've stayed consistent with our current `from_pretrained` method which handles everything for the user (e.g. placing the model on the correct the `device`). However, it goes a bit against PyTorch design (which would return an `nn.Module` to the user and have them apply compile themselves).

Reply:
I'd say that, for the purposes of repo, this is positive! 👍